samples/test/create-custom-job.test.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const {assert} = require('chai');
+const {after, describe, it} = require('mocha');
+const uuid = require('uuid').v4;
+const cp = require('child_process');
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+const cwd = path.join(__dirname, '..');
+
+const customJobDisplayName = `temp_create_custom_job_test${uuid()}`;
+const containerImageUri = 'gcr.io/ucaip-test/ucaip-training-test:latest';
+const project = process.env.CAIP_PROJECT_ID;
+const location = process.env.LOCATION;
+
+let customJobId;
+
+describe('AI platform create custom job', () => {
+  it('should create a new custom job', async () => {
+    const stdout = execSync(
+      `node ./create-custom-job.js ${customJobDisplayName} \
+                                     ${containerImageUri} \
+                                     ${project} ${location}`,
+      {
+        cwd,
+      }
+    );
+    assert.match(stdout, /Create custom job response/);
+    customJobId = stdout
+      .split('/locations/us-central1/customJobs/')[1]
+      .split('\n')[0]
+      .slice(0, 19);
+  });
+
+  after('should cancel the customJob and delete it', async () => {
+    execSync(
```

Comment:
I see a lot of cleanup that happens in these samples in the `after` hook (which is good).  Question - would it be possible to add a reaper of some sort that cleans up things that get missed?  Kokoro jobs will sometimes get killed, processes crash, and "stuff" just happens sometimes that leaves zombie resources.  When we don't have cleanup tasks that do timestamp based cleanup for leaked resources, we tend to start getting mysterious quota issues or slowdowns down the road.  It would be nice to be extra defensive here.

Reply:
Yep.